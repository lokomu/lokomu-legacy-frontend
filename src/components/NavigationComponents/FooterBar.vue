<template>
  <div
    :class="
      userStore.isLoggedIn
        ? 'relative bg-transparent'
        : 'absolute bg-white dark:bg-black'
    "
    class="left-0 w-full py-12 px-4 border-t border-gray-200 dark:border-gray-900"
  >
    <div class="mx-auto container flex flex-col items-center justify-center">
      <router-link
        to="/"
        class="mx-auto flex"
      >
        <LogoIcon class="h-10 fill-green-500 dark:fill-white" />
      </router-link>
      <h1 class="text-center text-pink-500 text-2xl font-bold pt-3">
        {{ $t("misc.slogan") }}
      </h1>
      <ul class="md:flex items-center my-6 text-base text-color">
        <li
          v-for="item in menuItems"
          :key="item.key"
          class="md:mr-6 cursor-pointer pt-4 lg:py-0"
        >
          <router-link
            :to="item.url"
            class="dark:text-white text-floyd-800 focus:underline hover:text-gray-500"
          >
            {{ $t(`misc.help.${item.key}`) }}
          </router-link>
        </li>
      </ul>
      <p class="text-floyd-900 text-sm text-color mb-16 f-f-l dark:text-white">
        {{ $t("misc.copyright") }}
      </p>
      <div
        class="flex mx-auto"
        :class="{ 'pb-20 md:pb-0': userStore.isLoggedIn }"
      >
        <LanguageSwitcher />
        <ThemeToggler class="ml-4" />
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import LogoIcon from "@/components/IconComponents/LogoIcon.vue";
import ThemeToggler from "@/components/BaseComponents/ThemeToggler.vue";
import LanguageSwitcher from "@/components/BaseComponents/LanguageSwitcher.vue";
import { useUserStore } from "@/stores/UserStore";

const userStore = useUserStore();

const menuItems = [
  { key: "about", url: "/about" },
  { key: "terms", url: "/terms" },
  { key: "help", url: "/help" },
  { key: "privacy", url: "/privacy" },
];
</script>
