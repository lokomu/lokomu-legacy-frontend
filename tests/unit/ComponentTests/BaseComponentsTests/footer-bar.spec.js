import { shallowMount } from "@vue/test-utils";
import FooterBar from "@/components/NavigationComponents/FooterBar.vue";
import { describe, expect, it} from "vitest";

describe("FooterBar component", () => {
  let wrapper = shallowMount(FooterBar);

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
