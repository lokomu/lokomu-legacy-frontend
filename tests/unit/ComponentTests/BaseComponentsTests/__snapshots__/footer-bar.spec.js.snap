// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FooterBar component > renders correctly 1`] = `
<div
  class="absolute bg-white dark:bg-black left-0 w-full py-12 px-4 border-t border-gray-200 dark:border-gray-900"
>
  <div
    class="mx-auto container flex flex-col items-center justify-center"
  >
    <a
      class="mx-auto flex"
    >
      <logo-icon-stub
        class="h-10 fill-green-500 dark:fill-white"
      />
    </a>
    <h1
      class="text-center text-pink-500 text-2xl font-bold pt-3"
    >
      misc.slogan
    </h1>
    <ul
      class="md:flex items-center my-6 text-base text-color"
    >
      
      <li
        class="md:mr-6 cursor-pointer pt-4 lg:py-0"
      >
        <a
          class="dark:text-white text-floyd-800 focus:underline hover:text-gray-500"
        >
          misc.help.about
        </a>
      </li>
      <li
        class="md:mr-6 cursor-pointer pt-4 lg:py-0"
      >
        <a
          class="dark:text-white text-floyd-800 focus:underline hover:text-gray-500"
        >
          misc.help.terms
        </a>
      </li>
      <li
        class="md:mr-6 cursor-pointer pt-4 lg:py-0"
      >
        <a
          class="dark:text-white text-floyd-800 focus:underline hover:text-gray-500"
        >
          misc.help.help
        </a>
      </li>
      <li
        class="md:mr-6 cursor-pointer pt-4 lg:py-0"
      >
        <a
          class="dark:text-white text-floyd-800 focus:underline hover:text-gray-500"
        >
          misc.help.privacy
        </a>
      </li>
      
    </ul>
    <p
      class="text-floyd-900 text-sm text-color mb-16 f-f-l dark:text-white"
    >
      misc.copyright
    </p>
    <div
      class="flex mx-auto"
    >
      <language-switcher-stub />
      <theme-toggler-stub
        class="ml-4"
      />
    </div>
  </div>
</div>
`;
